### PR TITLE
Add event types filters on the map.

### PR DESCRIPTION
**Pull request by hhamon**

### CODE REVIEW

File: src/AppBundle/Api/EventProvider.php [hunk 1]
```diff
@@ -16,11 +18,14 @@ public function __construct(EventRepository $repository, UrlGeneratorInterface $
         $this->urlGenerator = $urlGenerator;
     }
 
-    public function getUpcomingEvents(): array
+    public function getUpcomingEvents(string $type = null): array
```
Comment (HeahDude) [inline src/AppBundle/Api/EventProvider.php]: Please add a `@throws` phpdoc.

File: src/AppBundle/Api/EventProvider.php [hunk 2]
```diff
@@ -16,11 +18,14 @@ public function __construct(EventRepository $repository, UrlGeneratorInterface $
...
     {
-        $data = [];
+        if ($type && !EventCategories::getCategoryName($type)) {
+            throw new EventException(sprintf('Given event category "%s" is invalid.', $type));
+        }
 
-        foreach ($this->repository->findUpcomingEvents() as $event) {
+        $data = [];
```
Comment (HeahDude) [inline src/AppBundle/Api/EventProvider.php]: In PHP 7 a micro optim is to do:
```php
return $data ?? [];
```
and remove this memory call which might be needless.

File: src/AppBundle/Controller/Api/CommitteesController.php [hunk 3]
```diff
@@ -25,8 +27,12 @@ public function getApprovedCommitteesAction()
      * @Route("/events", name="api_committees_events")
      * @Method("GET")
      */
-    public function getUpcomingCommitteesEventsAction()
+    public function getUpcomingCommitteesEventsAction(Request $request)
     {
-        return new JsonResponse($this->get('app.api.event_provider')->getUpcomingEvents());
+        try {
+            return new JsonResponse($this->get('app.api.event_provider')->getUpcomingEvents($request->query->get('type')));
+        } catch (\Exception $e) {
```
Comment (HeahDude) [inline src/AppBundle/Controller/Api/CommitteesController.php]: Please catch `EventException` only, we don't want bad request code for anything else.

File: src/AppBundle/Api/EventProvider.php (hunk 1 above)
Comment (hhamon) [inline src/AppBundle/Api/EventProvider.php]: Fixed. Thanks.

File: src/AppBundle/Api/EventProvider.php (hunk 2 above)
Comment (hhamon) [inline src/AppBundle/Api/EventProvider.php]: Fixed. Thanks.

File: src/AppBundle/Controller/Api/CommitteesController.php (hunk 3 above)
Comment (hhamon) [inline src/AppBundle/Controller/Api/CommitteesController.php]: Fixed. Thanks.

File: src/AppBundle/Api/EventProvider.php [hunk 4]
```diff
@@ -16,11 +18,16 @@ public function __construct(EventRepository $repository, UrlGeneratorInterface $
...
     }
 
-    public function getUpcomingEvents(): array
+    /**
+     * @throws EventException
+     */
+    public function getUpcomingEvents(string $type = null): array
     {
-        $data = [];
+        if ($type && !EventCategories::getCategoryName($type)) {
```
Comment (HeahDude) [inline src/AppBundle/Api/EventProvider.php]: Why testing it? Maybe this should not be nullable in the first place?

Comment (hhamon) [inline src/AppBundle/Api/EventProvider.php]: because `null` means I want to all events no matter the category is.

**Pull request merged**